File: ClientApp/src/app/shared/models/register-model.ts

export class RegisterModel {
    name!: string;
    socialNumber!: number;
    cifNumber!: number;
    email!: string;
    password!: string;
    dateCreated!: Date;
    isAdmin!: boolean;
    isConfirmed: boolean = false;
    isValid: boolean = false;
}

File: Model/RegisterModel.cs
using coop2._0.Entities;
using System;

namespace coop2._0.Model
{
    public class RegisterModel
    {
        public string Name { get; set; }
        public string SocialNumber { get; set; }
        public string Email { get; set; }
        public string Password { get; set; }
        public DateTime DateCreated { get; set; }
    }
}